microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/WithSpanInterceptor.java

```diff
@@ -75,19 +78,42 @@ public Object interceptSpan(InvocationContext context) throws Exception {
             spanName = className + "." + method.getName();
         }
 
-        // Start new Span
-        Span span = tracer.spanBuilder(spanName)
+        SpanBuilder spanBuilder = tracer.spanBuilder(spanName)
                 .setSpanKind(annotation.kind())
-                .setParent(Context.current())
-                .startSpan();
+                .setParent(Context.current());
+
+        for (int i = 0; i < method.getParameters().length; i++) {
+            Parameter p = method.getParameters()[i];
+            if (p.isAnnotationPresent(SpanAttribute.class)) {
+                SpanAttribute spanAttribute = p.getAnnotation(SpanAttribute.class);
+                var attrName = spanAttribute.value().isBlank() ? p.getName() : spanAttribute.value();
+                Class<?> paramType = p.getType();
+                Object pValue = context.getParameters()[i];
+
+                if (paramType.isAssignableFrom(String.class)) {
+                    spanBuilder.setAttribute(attrName, (String) pValue);
+                } else if (paramType.isAssignableFrom(long.class) || paramType.isAssignableFrom(Long.class)) {
+                    spanBuilder.setAttribute(attrName, (long) pValue);
+                } else if (paramType.isAssignableFrom(double.class) || paramType.isAssignableFrom(Double.class)) {
+                    spanBuilder.setAttribute(attrName, (double) pValue);
+                } else if (paramType.isAssignableFrom(boolean.class) || paramType.isAssignableFrom(Boolean.class)) {
+                    spanBuilder.setAttribute(attrName, (boolean) pValue);
+                } else {
+                    spanBuilder.setAttribute(attrName, pValue.toString());
+                }
```

Comment:
I'm pretty sure these are backwards. You want `String.class.isAssignableFrom(paramType)`. Or you want to do something like:
```
switch (paramType) {
case Class<?> x when Long.class.isAssignableFrom(paramType): // this may? work for primitives too? can't remember if autoboxing kicks in?
  spanBuilder.setAttribute(attrName, x.cast(pValue));
  break;
case Class<?> x when Double.class.isAssignableFrom(paramType):
  spanBuilder.setAttribute(attrName, x.cast(pValue));
  break;
case Class<?> x when Boolean.class.isAssignableFrom(paramType):
  spanBuilder.setAttribute(attrName, x.cast(pValue));
  break;
default:
  spanBuilder.setAttribute(attrName, pValue.toString());
  break;
}
```